include/zephyr/bluetooth/audio/has.h

```diff
@@ -470,6 +470,15 @@ static inline int bt_has_preset_active_clear(void)
  */
 int bt_has_preset_name_change(uint8_t index, const char *name);
 
+/**
+ * @brief Change the Hearing Aid Features.
+ *
+ * Change the hearing aid features.
+ *
+ * @param features The features to be set.
+ */
+void bt_has_features_set(uint8_t features);
```

Comment:
I am not so convinced about using `uint8_t features` here. It's because the value consist of:
![image](https://github.com/zephyrproject-rtos/zephyr/assets/10126877/b320b58f-8042-428c-b887-60b0651e3eab)
So this gives a backdoor to set eother RFU values, mark `Writable Presets Support` and `Dynamic Presets` etc even if `BT_HAS_PRESET_SUPPORT` is disabled, mark `Preset Synchronization Support` even if not binaural, etc.
This makes the API not user-proof

Reply:
Yes, my bad. I will update and take this into consideration.

Reply:
But on that matter, would you prefer a uint8_t mask that is checked, or actually providing each bit as an argument?

Reply:
I guess that you could rename and reuse `struct bt_has_register_param` which basically initaites the feature values

Reply:
> I guess that you could rename and reuse `struct bt_has_register_param` which basically initaites the feature values

I concur :) 

Another alternative is (which we do for the VCP Volume Renderer) is that we have a single struct for registering all the initial values, and then have individual functions to set the others. So in this case, that would be a `bt_has_set_type(enum bt_has_hearing_aid_type type)`, `bt_has_set_preset_sync_support(bool preset_sync_support)` and `bt_has_set_independent_presets(bool independent_presets)`

The difference between VCS and HAS here is of course it's a single bitfield in HAS, but different characteristics in VCS. 

The downside of reusing the `register_param` is that it it's harder to expand those parameters later if the HAS spec changes. The downside of my suggested alternative is that more functions -> More ROM usage

Reply:
I have implemented it as reusing the param struct, and I was thinking of keeping it like this unless the reviewers are in agreement that the other approach is better? :) @Thalley @MariuszSkamra

Reply:
Indeed, it may change as part of https://github.com/zephyrproject-rtos/zephyr/issues/57454